docs - qualify some resource-queue specific content (part 1)

lisakowen opened the pull request:
start qualifying resource queue- and resource group- specific content - PART 1, more to come

- s3 page - remove reference to gp_vmem_protect_limit, keep generic memory reference
- Greenplum Database Memory Overview page - qualify RQ gp_vmem and overcommit_ratio discussion, add RG overcommit_ratio statement, qualify RQ and add RG segment memory allocation, add segment memory calculations
- best practices (3 pages)- qualify gp_vmem_protect_limit references as RQ, add RQ to some section names
- gp_vmem_protect_limit, statement_mem, runaway_detector_activation_percent, max_statement_mem - add note that these are RQ-only
- install_guide - qualify vm.overcommit_ratio statement as RQ, add RG-specific statement

links to some of the updates on the doc review site:
- http://docs-gpdb-review-staging.cfapps.io/510/admin_guide/wlmgmt_intro.html
- http://docs-gpdb-review-staging.cfapps.io/510/best_practices/workloads.html
- http://docs-gpdb-review-staging.cfapps.io/510/ref_guide/config_params/guc-list.html#gp_vmem_protect_limit

dyozie commented on gpdb-doc/dita/admin_guide/wlmgmt_intro.xml:
```diff
@@ -104,20 +104,15 @@
-        <note>The resource management scheme currently active (resource queues or resource
-          groups) for your Greenplum Database cluster may impose additional considerations
-          on <varname>gp_vmem_protect_limit</varname>.</note> 
+          failure. If <varname>gp_vmem_protect_limit</varname> is set too high, queries can fail.</p>
```
This is good, but I think it would be helpful to also add at the end "gp_vmem_protect_limit does not affect segment memory when using resource groups for resource management" or something similar.

lisakowen commented:
qualify all RQ and RG gucs and system tables/views.

pengzhout commented on gpdb-doc/dita/ref_guide/config_params/guc-list.xml:
```diff
@@ -7261,6 +7269,7 @@
   <topic id="max_resource_groups">
     <title>max_resource_groups</title>
     <body>
+      <note>The <codeph>max_resource_groups</codeph> server configuration parameter is enforced only when resource group-based resource management is active.</note>
```
@gaos1 Have we removed this GUC?

gaos1 replied:
yes, we removed it.  according to this doc pr, it is already removed from doc. @lisakowen 
https://github.com/greenplum-db/gpdb/commit/042b6b5d5f8ebce45cb9feaf3bc0c14a3444ee55

lisakowen replied:
hmmm, thanks for noticing this.  i will look through the commits and see how that guc got added back in ...

pengzhout commented on gpdb-doc/dita/admin_guide/wlmgmt_intro.xml:
```diff
@@ -108,13 +108,23 @@
+          divided by the number of segments. Use the following formula to calculate a safe
+          value for segment memory when using resource groups for resource management.</p>
+        <p><codeblock>
+rg_perseg_mem = ((RAM * (vm.overcommit_ratio / 100) + SWAP) * gp_resource_group_memory_limit) / number_of_segments </codeblock></p>
```
Better to clarify that number_of_segments meaning the number per host.

lisakowen replied:
should RG calculation have the same denominator as that of gp_vmem_protect_limit calculation (i.e. max_acting_primary_segments)?

pengzhout replied:
Actually, it should not be the same as max_acting_primary_segments because, in RG, rg_perseg_mem can be different between hosts, each host may have a different number of segments configured, number_of_segments is calculated automatically and is the exact number on each host.

gaos1 replied:
@lisakowen it is different. in resource group, the denominator is active primary segments.

lisakowen commented:
@dyozie - addressed reviewer comments.  not sure if you have reviewed the new additions.  now qualifying RG gucs and system tables/views.  added an RG segment memory calculation in the memory overview page.

let me know if you have further edits.

lisakowen commented:
np
